imports/plugins/included/simple-inventory/server/no-meteor/mutations/updateSimpleInventoryBulk.js

```diff
@@ -0,0 +1,147 @@
+import SimpleSchema from "simpl-schema";
+import Random from "@reactioncommerce/random";
+import Logger from "@reactioncommerce/logger";
+import { ProductConfigurationSchema, SimpleInventoryCollectionSchema } from "../simpleSchemas";
+
+const logCtx = { name: "core-inventory" };
+
+const inputSchema = new SimpleSchema({
+  productConfiguration: ProductConfigurationSchema,
+  canBackorder: {
+    type: Boolean,
+    optional: true
+  },
+  inventoryInStock: {
+    type: SimpleSchema.Integer,
+    min: 0,
+    optional: true
+  },
+  isEnabled: {
+    type: Boolean,
+    optional: true
+  },
+  lowInventoryWarningThreshold: {
+    type: SimpleSchema.Integer,
+    min: 0,
+    optional: true
+  },
+  shopId: String
+});
+
+const updateFields = [
+  "canBackorder",
+  "inventoryInStock",
+  "isEnabled",
+  "lowInventoryWarningThreshold"
+];
+
+const defaultValues = {
+  canBackorder: false,
+  inventoryInStock: 0,
+  isEnabled: false,
+  lowInventoryWarningThreshold: 0
+};
+
+// eslint-disable-next-line require-jsdoc
+export default async function updateSimpleInventoryBulk(context, input) {
+  const { appEvents, collections } = context;
+  const { SimpleInventory } = collections;
+
+  const bulkOperations = input.updates.map((update) => {
+    const { productConfiguration, shopId } = update;
+    inputSchema.validate(update); // try/catch
+    const $set = { updatedAt: new Date() };
+    const $setOnInsert = {
+      "_id": Random.id(),
+      "createdAt": new Date(),
+      // inventoryReserved is calculated by this plugin rather than being set by
+      // users, but we need to init it to some number since this is required.
+      // Below we update this to the correct number if we inserted.
+      "inventoryReserved": 0,
+      // The upsert query below has only `productVariantId` so we need to ensure both are inserted
+      "productConfiguration.productId": productConfiguration.productId
+    };
+    updateFields.forEach((field) => {
+      const value = update[field];
+      if (value !== undefined && value !== null) {
+        $set[field] = value;
+      } else {
+        // If we are not setting the value here, then we add it to the setOnInsert.
+        // This is necessary because all fields are required by the collection schema.
+        $setOnInsert[field] = defaultValues[field];
+      }
+    });
+
+    // if (Object.getOwnPropertyNames($set).length === 1) {
+    //   throw new ReactionError("invalid-param", "You must provide at least one field to update.");
+    // }
```

Comment:
For this check, you could maybe keep it but just log it and `return null` instead of throwing? Then you'll need to filter out `null` from `bulkOperations` before the `bulkWrite`.